### ISSUE TITLE
New Formats (wishlist)

### ISSUE THREAD

**Issue by kholia** [magnumripper/JohnTheRipper#359]
- [x] Improve DMG formats (sparsebundle, sparseimage)
- [x] Fast and better support for fast hashes (sayantan, all)
- [x] Improve 7z support (dhiru, all)
- [x] MS OneNote (2003 to 2013) http://www.microsoft.com/en-in/download/details.aspx?id=14565 (OfficeFileFormatsProtocols.zip)
- [x] MS Windows EFS https://bitbucket.org/jmichel/dpapick
- [x] MS Windows BitLocker https://github.com/libyal/libbde (Elena has a PR #2427)
- [x] MS Access >= 2007 (ACCDB)
- [x] MS Access (2003 to 2007, Google "DocRecrypt")
- [x] BestCrypt 8.0 (JBC) http://www.jetico.com/bcmac-help/AB_containerfile.html http://www.jetico.com/linux/bcrypt-help/b_cntff.htm

**Comment by lukimausi** [comment]
Hello everyone, 
I wanna apologize in advance if i am posting in a wrong thread or something already discussed. Actually I know other people have had the my problem already, but i dont get around it by their explanations.
I revently found a old password protected .dmg file on one of my USB-Sticks. I have no idea what the pw could be, of course i tried all my usall suspects but nothing. I searced around as mentioned and it seems to be that John the Ripper is able to do the job, but since I am a total complete noob and computer-retard i dont understand the tutorials given on sites like open wall and others since the all assume the readers have some basic to advanced knowledge of using jtr and the windows cmd programm ( i have a windows 7 by know, no mac anymore). I tried hard to follow them, but failed and the worst part is, that i have no idea why it doesnt work.
So if i knew how to actually do it, my plan would be to download a right version of jtr, extract it, open it via the "terminal" of windows ( what was it called again?) tell it to extract the hash of my file, crack that hash, and get to my old documents, fotos etc in that file. I knwo also that there is an option to create your own dictionary for jtr, to narrow down the possible password try outs, but here as well, i have no clue of how to actually to it. 
So what i wish for would be a fool-proof step by step guide, for opening a .dmg file with a windows laptop.
thanks
luki
Thank you

**Comment by kholia** [comment]
BitLocker stuff is in progress now (https://github.com/kholia/libbde).

**Comment by DeusExMachina** [comment]
Voting for BestCrypt (.jbc) container.

**Comment by kholia** [comment]
https://www.jetico.com/linux/BestCrypt-2.0.10.tar.gz (has source code which is great).

**Comment by kholia** [comment]
https://www.symantec.com/connect/downloads/symantec-pgp-desktop-peer-review-source-code might be helpful for attacking PGP WDE.

**Comment by kholia** [comment]
PR https://github.com/magnumripper/JohnTheRipper/pull/2448 adds supports for Jetico BestCrypt.

**Comment by kholia** [comment]
@stephanschielke

JtR jumbo (this repository) now supports cracking password protected Jetico BestCrypt containers (.jbc). Can you give it a try, and provide some feedback? Thanks.

**Comment by kholia** [comment]
Removed `Revelation Password Manager` from the list. It hasn't been updated in years, and no one seems to care about it.

**Comment by kholia** [comment]
Writing a format for `The Bat!` requires significant reverse engineering effort.  `The Bat!` changes its "KDF" quite frequently. Also, non one seems to care about it.

**Comment by kholia** [comment]
Moved `PGP WDE`, KeePass GPU, SSH GPU, PFX GPU items to new issues.

**Comment by kholia** [comment]
MS Money support can be added with help from following projects.

* http://jackcessencrypt.sourceforge.net/

* http://jackcess.sourceforge.net/

**Comment by kholia** [comment]
http://jessekornblum.com/presentations/dodcc09.pdf is a good read for BitLocker stuff.
